sql/rowexec/insert.go

```diff
@@ -310,6 +337,106 @@ func (i *insertIter) getAutoIncVal(row sql.Row) int64 {
 	return autoIncVal
 }
 
+// findUuidPrimaryKey searches for a column in the primary key that is a UUID column. UUID columns are defined as
+// either a char(36) or varchar(36) that has a column default of UUID() or a binary(16) or varbinary(16) that has
+// a column default of UUID_to_bin(UUID()). This is meant to be similar to auto_increment rules, which allow only
+// one auto_increment column that must be part of a key. If a matching column is found, its index is returned,
+// otherwise -1 is returned to indicate that no matching column was found.
+func findUuidPrimaryKey(schema sql.Schema) int {
```

Comment:
Want to restrict this match only for a single column key, or no?

Reply:
My thinking was to mirror `auto_increment` requirements as much as possible with this. So, since an `auto_increment` column can be a part of a composite primary key, I think we should allow an automatic UUID column to be used in a composite key, too. I also think it'll be useful for customers to have additional PK fields.